app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/SurroundWithMiniPlayer.kt

```diff
@@ -58,12 +75,155 @@ fun SurroundWithMiniPlayer(displayPlayer: Boolean, screen: @Composable () -> Uni
               onPauseClick = { viewModel.pause() },
               progress = progress.floatValue)
         } else {
-          PlayerDragHandle()
-          Row(
-              modifier = Modifier.fillMaxSize(),
-              horizontalArrangement = Arrangement.Center,
-              verticalAlignment = Alignment.CenterVertically) {
-                Box(modifier = Modifier.size(250.dp).background(Color.LightGray))
+          Column(
+              modifier = Modifier.fillMaxSize().padding(bottom = 84.dp),
+              horizontalAlignment = Alignment.CenterHorizontally,
+              verticalArrangement = Arrangement.SpaceBetween) {
+                PlayerDragHandle()
+                Row(
+                    horizontalArrangement = Arrangement.SpaceAround,
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier.fillMaxWidth()) {
+                      IconButton(onClick = { /*TODO*/}) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.broadcast_icon),
+                            contentDescription = "",
+                            tint = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.size(50.dp))
+                      }
+                      IconButton(onClick = { /*TODO*/}) {
+                        Icon(
+                            painter = painterResource(id = R.drawable.beacon_add_icon),
+                            contentDescription = "",
+                            tint = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.size(50.dp))
+                      }
```

Comment:
I would advise creating a function for displaying an iconButton to improve modularity of your code. It avoids repeating the same lines multiple times